File: python/sieve.py
def sieve(n):
    # create a list of non-prime numbers
    data = [False for x in range(n)]

    print(1, end=" ")
    # go through all the numbers
    for i in range(2, n):
        # if the number is prime, print it
        if not data[i]:
            print(i, end=" ")
        # set all multiples of the number as not prime
        for j in range(i, n, i):
            data[j] = True
    print()

from sys import argv

# set the default value
n = 100
if len(argv) == 2:
    # read the number to check for primes under
    n = int(argv[1])
print("Primes from 1 to", n)
sieve(n)

File: c/sieve.c
#include<stdio.h>
#include<stdlib.h>

void init(int n, char* data, char value){
	for(int i = 0; i < n; i++)
		data[i] = value;
}

void sieve(int n, char* data){
	init(n, data, 0);
	
	for(int i = 2; i < n; i++){
		if(data[i] == 0)
			printf("%d ", i);
		for(int j = i; j < n; j = j + i)
			data[j] = 1;
	}

	printf("\n");
}

int main(int argc, char **argv){
	int n = 100;
	if(argc == 2)
		n = atoi(argv[1]);

	char* data = malloc(sizeof(char) * n);
	
	printf("Primes from 1 to %d\n", n);

	sieve(n, data);

	free(data);

	return 0;
}


